### PR TITLE
QA Automation test task - mass user registration load test

### PR DESCRIPTION
**Pull request by inpv**
### Motivation and context

Since I haven't found any particular load/performance tests in the repo, I decided to start making them myself. I think `locust` can be a useful instrument for load testing, since it's developer-friendly and easily integrated into Python code.
In this particular example I start a locust env with a runner and greenlets and start swarming the endpoint with `POST` requests from `api_client.auth_api.create_register` populated by `Faker` and `secrets` data, measuring the average response time and asserting for 0 failures before closing the server instance.

### How has this been tested?

This test has been launched on a (custom) Arch Linux dev env, using python3.9, pytest 6.2.5 and locust 2.13.0.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by inpv** [comment]
Well, I have a couple questions:

1. Some of the checks fail because `faker` and `locust` aren't in the default requirements list. May I add them as part of this PR as well or should I open another PR for this?

2. Pylint seems to mark the check upon finding an "unnecessary lambda" warning [here](https://github.com/inpv/cvat/blob/fc399e8915bbd5ae8dbf40d6317ee2d50d63a72b/tests/python/rest_api/test_load_register.py#L140) as failed, however rewriting the function like it "should" be:
```Python
def quit_runner(environment)
    return environment.runner.quit()

# And then, inside the main function
gevent.spawn_later(10, quit_runner(environment=env))
```
doesn't get to start the default runner at all (response data at this [line](https://github.com/inpv/cvat/blob/fc399e8915bbd5ae8dbf40d6317ee2d50d63a72b/tests/python/rest_api/test_load_register.py#L113) isn't printed, only with lambda warning suppressed). I guess I'll disable the warning in this test for now: `# pylint: disable=unnecessary-lambda`, until I find a more compatible runner.

**Comment by kirill-sizov** [comment]
> Well, I have a couple questions:
> 
> 1. Some of the checks fail because `faker` and `locust` aren't in the default requirements list. May I add them as part of this PR as well or should I open another PR for this?
> 2. Pylint seems to mark the check upon finding an "unnecessary lambda" warning [here](https://github.com/inpv/cvat/blob/fc399e8915bbd5ae8dbf40d6317ee2d50d63a72b/tests/python/rest_api/test_load_register.py#L140) as failed, however rewriting the function like it "should" be:
> 
> ```python
> def quit_runner(environment)
>     return environment.runner.quit()
> 
> # And then, inside the main function
> gevent.spawn_later(10, quit_runner(environment=env))
> ```
> 
> doesn't get to start the default runner at all (response data at this [line](https://github.com/inpv/cvat/blob/fc399e8915bbd5ae8dbf40d6317ee2d50d63a72b/tests/python/rest_api/test_load_register.py#L113) isn't printed, only with lambda warning suppressed). I guess I'll disable the warning in this test for now: `# pylint: disable=unnecessary-lambda`, until I find a more compatible runner.

Generally, you can feel free to do everything to make PR "green", if disabling of some checks seems unfounded for us, we will mention it in review.
1) Update tests requirements.txt  so that CI can run your test
2) In case with this with pylint warning it's looks fine to me to disable it.

File: tests/python/rest_api/test_load_register.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            try:
+                # forming the request body
+                register_request = RegisterSerializerExRequest(
+                    username=username,
+                    password1=passwd,
+                    password2=passwd,
+                    email=email,
+                    first_name=first_name,
+                    last_name=last_name,
+                )
```
Comment (kirill-sizov) [inline tests/python/rest_api/test_load_register.py:111]: When I run the test locally, I see some requests fail with 400, but the test doesn't fail. I think we should implement this test in such a way that we don't have this error.

**Review by kirill-sizov — CHANGES_REQUESTED** [review]

File: tests/python/rest_api/test_load_register.py (hunk 1 above)
Comment (inpv) [reply tests/python/rest_api/test_load_register.py:111]: Do you want to catch the request exceptions and make the test fail upon parsing a non-200 response, or to suppress errors when sending the request altogether?

Comment (inpv) [reply tests/python/rest_api/test_load_register.py:53]: Ended up doing both

Comment (kirill-sizov) [reply tests/python/rest_api/test_load_register.py:53]: In my opinion the should fail if server return non-200 responses.
We are trying to check the average user registration time, but how can we trust this number if some (or all) of our registrations fail.

Comment (inpv) [reply tests/python/rest_api/test_load_register.py:53]: Sure, thought so too. Corrected as needed.

**Comment by inpv** [comment]
@kirill-sizov For some reason CI api tests take forever to load. My assumption is that it may have something to do with `gevent` coroutines. Can you check the logs, please? I can run the Actions locally with [act](https://github.com/nektos/act), but it'll take some time to setup.

**Comment by inpv** [comment]
@kirill-sizov Right now the test cannot pass the CI `rest_api` checks because locust uses a monkey patching `gevent` call in its own [code](https://github.com/locustio/locust/blob/dcfa746e85ae03bc00ec18b3105f79076eef2ca7/locust/__init__.py#L11), which is executed **after** `urllib3` has already loaded the inbuilt Python `ssl` module:
```
.env/lib/python3.9/site-packages/locust/__init__.py:11
  ~/Development/cvat/.env/lib/python3.9/site-packages/locust/__init__.py:11: MonkeyPatchWarning: Monkey-patching ssl after ssl has already been imported may lead to errors, including RecursionError on Python 3.6. It may also silently lead to incorrect behaviour on Python 3.7. Please monkey-patch earlier. See https://github.com/gevent/gevent/issues/1016. Modules that had direct imports (NOT patched): ['urllib3.util.ssl_ (~/Development/cvat/.env/lib/python3.9/site-packages/urllib3/util/ssl_.py)', 'urllib3.util (~/Development/cvat/.env/lib/python3.9/site-packages/urllib3/util/__init__.py)']. 
    monkey.patch_all()
```
Which leads to infinite recursion during CI checks making them take too long to complete:
```
tests/python/rest_api/test_resource_import_export.py:102: in _idempotent_saving_resource_to_cloud_storage
    _save_resource_to_cloud_storage(*args, **kwargs)
tests/python/rest_api/test_resource_import_export.py:75: in wrapper
    client = define_client()
tests/python/rest_api/test_resource_import_export.py:45: in define_client
    s3 = boto3.resource(
.env/lib/python3.9/site-packages/boto3/__init__.py:102: in resource
    return _get_default_session().resource(*args, **kwargs)
.env/lib/python3.9/site-packages/boto3/session.py:384: in resource
    client = self.client(
.env/lib/python3.9/site-packages/boto3/session.py:258: in client
    return self._session.create_client(
.env/lib/python3.9/site-packages/botocore/session.py:847: in create_client
    client = client_creator.create_client(
.env/lib/python3.9/site-packages/botocore/client.py:86: in create_client
    client_args = self._get_client_args(
.env/lib/python3.9/site-packages/botocore/client.py:355: in _get_client_args
    return args_creator.get_client_args(
.env/lib/python3.9/site-packages/botocore/args.py:99: in get_client_args
    endpoint = endpoint_creator.create_endpoint(
.env/lib/python3.9/site-packages/botocore/endpoint.py:293: in create_endpoint
    http_session = http_session_cls(
.env/lib/python3.9/site-packages/botocore/httpsession.py:203: in __init__
    self._manager = PoolManager(**self._get_pool_manager_kwargs())
.env/lib/python3.9/site-packages/botocore/httpsession.py:222: in _get_pool_manager_kwargs
    'ssl_context': self._get_ssl_context(),
.env/lib/python3.9/site-packages/botocore/httpsession.py:231: in _get_ssl_context
    return create_urllib3_context()
.env/lib/python3.9/site-packages/botocore/httpsession.py:77: in create_urllib3_context
    context.options |= options
/usr/lib/python3.9/ssl.py:603: in options
    super(SSLContext, SSLContext).options.__set__(self, value)
/usr/lib/python3.9/ssl.py:603: in options
    super(SSLContext, SSLContext).options.__set__(self, value)
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
```
The overall monkeypatching was [introduced](https://github.com/locustio/locust/issues/569) already in version 0.8, so downgrading is not an option, their Python is outdated.
Everything else checks fine. I'll continue investigating, but I already consider moving to another load testing tool at this point.

**Pull request closed without merging**